packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v0/mod.rs

```diff
@@ -465,10 +511,8 @@ fn insert_values(
                                 ),
                             }
                         } else {
-                            return Err(ProtocolError::DataContractError(
-                                DataContractError::InvalidContractStructure(
-                                    "byteArray should always be true if defined".to_string(),
-                                ),
+                            return Err(DataContractError::InvalidContractStructure(
```

Comment:
You are returning non consensus error here because we already supposed to validate with JSON Schema?

Reply:
Are you sure all this errors are related to consensus?

For example:
 Self::ContractError(DataContractError::CorruptedCodeExecution(_)) => 1092, 
 
 Did you move the rest of basic errors related data contracts under contract errors section?